### PR TITLE
Add permissions to get secrets to pv-controller.

### PR DESCRIPTION
**Pull request by jsafrane**
Glusterfs and Ceph provisioners needs to read any(!) secret on the system.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1388316

### CODE REVIEW

**Comment by jsafrane** [comment]
[test]

File: pkg/cmd/server/bootstrappolicy/infra_sa_policy.go [hunk 1]
```diff
@@ -589,11 +589,16 @@ func init() {
...
-				// Gluster provisioner
+				// Glusterfs provisioner
 				{
 					Verbs:     sets.NewString("get", "create", "delete"),
 					Resources: sets.NewString("services", "endpoints"),
 				},
+				// Glusterfs & Ceph provisioner
+				{
+					Verbs:     sets.NewString("get"),
+					Resources: sets.NewString("secrets"),
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/infra_sa_policy.go]: what's the status on the work to mitigate this exposure?
* only work off of storageclass objects, not PV annotations
* require the targeted secret to be a specific type, indicating intent on the secret-creator's part to have it be used by the provisioner?

Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/infra_sa_policy.go]: opened https://github.com/kubernetes/kubernetes/pull/35675 to require the targeted secret to be a specific type

Comment (jsafrane) [inline pkg/cmd/server/bootstrappolicy/infra_sa_policy.go]: > only work off of storageclass objects, not PV annotations

Yes, StorageClass only. Kubernetes master (upcoming 1.5) does not use PV annotations and  all the necessary patches were ported to Origin 1.4. I don't think we used PV annotations in "vanilla" Kubernetes 1.4.

Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/infra_sa_policy.go]: can you pick 35675 as part of this, and keep an eye on new PV provisioners upstream to make sure they follow that restriction?

Comment (jsafrane) [inline pkg/cmd/server/bootstrappolicy/infra_sa_policy.go]: sure, I'll incorporate 35675 into this PR it when it's merged.

**Comment by jsafrane** [comment]
Added https://github.com/kubernetes/kubernetes/pull/35675 and tested locally.

**Comment by liggitt** [comment]
Just noticed your comment about ceph provisioners. I didn't catch that in my upstream pull. Is that using a different method to look up secrets? If so, can you unify that with the others and add the type restriction for PV lookup?

**Comment by jsafrane** [comment]
> Just noticed your comment about ceph provisioners. I didn't catch that in my upstream pull. Is that using a different method to look up secrets? If so, can you unify that with the others and add the type restriction for PV lookup?

Not sure what you mean, Ceph is already handled well in this PR:

https://github.com/openshift/origin/pull/11581/files#diff-713034e6da86ab87b08bd96b86c77b01R284

`pkg/volume/rbd` = Ceph RBD plugin

**Comment by liggitt** [comment]
huh, I was confused by the presence of both cephfs and rbd packages upstream

**Comment by liggitt** [comment]
thanks. regarding the bz comment ('this not really *required* in Origin in 3.4'), can this wait until we pick up a rebase post-1.4?

**Comment by jsafrane** [comment]
> can this wait until we pick up a rebase post-1.4?


No, I'd like it in 1.4 as it is a prerequisite for CNS (https://www.redhat.com/en/technologies/storage/use-cases/container-native-storage)

**Comment by liggitt** [comment]
if we've already picked back changes that make the provisioner read secrets, then I guess I want the type restriction as well, so LGTM

**Comment by liggitt** [comment]
#10773 flake
[test]

**Comment by liggitt** [comment]
[merge]

**Comment by jsafrane** [comment]
[test]

**Comment by jsafrane** [comment]
[test]

**Comment by jsafrane** [comment]
[test]

**Comment by jsafrane** [comment]
[test]

**Comment by childsb** [comment]
[test]

**Comment by childsb** [comment]
#10773 flake
[test]

**Pull request merged**